### PR TITLE
Up 157 teacher can add exercise to cohort

### PR DESCRIPTION
**Pull request by Alex90Jennings**

### CODE REVIEW

File: src/controllers/cohort.js [hunk 1]
```diff
@@ -51,3 +52,37 @@ export const getCohortById = async (req, res) => {
...
+  } catch (e) {
+    return sendMessageResponse(res, 500, 'Unable to get exercise')
+  }
+}
+
+export const createCohortExercise = async (req, res) => {
+  const { exerciseId: exerciseID, cohortId: cohortID } = req.body
+
+  try {
+    const createdCohortExercise = await CohortExercise.createdCohortExercise(
```
Comment (dearshrewdwit) [inline src/controllers/cohort.js:79]: This looks like a typo - should it not be `createCohortExercise`, not `createdCohortExercise`?

File: src/controllers/cohort.js [hunk 2]
```diff
@@ -51,3 +52,37 @@ export const getCohortById = async (req, res) => {
...
+      }
+    })
+    return sendDataResponse(res, 200, { cohortExercises: formattedExercise })
+  } catch (e) {
+    return sendMessageResponse(res, 500, 'Unable to get exercise')
+  }
+}
+
+export const createCohortExercise = async (req, res) => {
+  const { exerciseId: exerciseID, cohortId: cohortID } = req.body
```
Comment (dearshrewdwit) [inline src/controllers/cohort.js:76]: What's the reasoning to map the keys to different variables? Can you not work with `exerciseId` and `cohortId` ?

File: src/controllers/cohort.js [hunk 3]
```diff
@@ -51,3 +52,37 @@ export const getCohortById = async (req, res) => {
...
   }
 }
+
+export const getAllCohortExercises = async (req, res) => {
+  const cohortId = Number(req.params.id)
+  const whereData = {}
+  whereData.cohortId = cohortId
+
+  try {
+    const foundExercise = await CohortExercise.findAll({ whereData })
```
Comment (dearshrewdwit) [inline src/controllers/cohort.js:62]: I expect variables containing arrays to be named using plural nouns - so `foundExercises`

File: src/controllers/cohort.js [hunk 4]
```diff
@@ -51,3 +52,37 @@ export const getCohortById = async (req, res) => {
...
+
+    const formattedExercise = foundExercise.map((exercise) => {
```
Comment (dearshrewdwit) [inline src/controllers/cohort.js:64]: and so: `formattedExercises`

File: src/domain/cohortExercise.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    })
+    return CohortExercise.fromDb(createdExercise)
+  }
+
+  static async findAll({ whereData }) {
+    return CohortExercise._findMany({ whereData })
+  }
+
+  static async _findMany({ whereData }) {
+    const foundExercise = await dbClient.CohortExercise.findMany({
```
Comment (dearshrewdwit) [inline src/domain/cohortExercise.js:38]: Rename = plural noun

File: src/routes/cohort.js [hunk 6]
```diff
@@ -10,5 +16,16 @@ const router = Router()
 router.post('/', validateAuthentication, validateTeacherRole, create)
 router.get('/', validateAuthentication, validateTeacherRole, cohorts)
 router.get('/:id', validateAuthentication, validateTeacherRole, getCohortById)
+router.get(
+  '/:id/cohortExercises',
+  validateAuthentication,
+  getAllCohortExercises
+)
+router.post(
+  '/:id/cohortExercise',
```
Comment (dearshrewdwit) [inline src/routes/cohort.js:25]: update path to use plural resource: `'/:id/cohortExercises'`

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/controllers/cohort.js (hunk 1 above)
Comment (Alex90Jennings) [reply src/controllers/cohort.js:79]: corrected

File: src/controllers/cohort.js (hunk 2 above)
Comment (Alex90Jennings) [reply src/controllers/cohort.js:76]: exerciseId: exerciseId was not needed, so I have left it as {exerciseId, cohortId} = req.body

File: src/controllers/cohort.js (hunk 3 above)
Comment (Alex90Jennings) [reply src/controllers/cohort.js:62]: done

File: src/routes/cohort.js (hunk 6 above)
Comment (Alex90Jennings) [reply src/routes/cohort.js:25]: done

File: src/controllers/cohort.js (hunk 4 above)
Comment (Alex90Jennings) [reply src/controllers/cohort.js:64]: done

**Pull request merged**